src/pages/index.js

```diff
@@ -82,9 +47,173 @@ const addPlaceModal = new PopupWithForm(
   handleNewPlaceSubmit
 );
 
+const updateAvatarModal = new PopupWithForm(
+  "#avatar-update-modal",
+  handleAvatarSubmit
+);
+const imagePreviewModal = new PopupWithImage("#places-preview-modal");
+const deletePlaceModal = new PopupWithForm("#places-delete-modal", () => {});
+
 editProfileModal.setEventListeners();
 imagePreviewModal.setEventListeners();
+updateAvatarModal.setEventListeners();
 addPlaceModal.setEventListeners();
-section.renderItems();
+
+deletePlaceModal.setEventListeners();
+avatarValidation.enableValidation();
 profileEditValidation.enableValidation();
 addPlaceValidation.enableValidation();
+
+function handleSubmit(
+  request,
+  popupInstance,
+  loadingText = "Saving...",
+  defaultText = "Save",
+  disableBtn = false
+) {
+  popupInstance.setLoading(true, loadingText);
+
+  return request()
+    .then(() => {
+      popupInstance.close();
+      popupInstance.reset();
+      if (disableBtn) {
+        popupInstance.setLoading(true);
+      } else {
+        popupInstance.setLoading(false);
+      }
+    })
+    .catch((error) => {
+      console.error("Error:", error);
+      disableBtn = false;
+    })
+    .finally(() => {
+      popupInstance.setLoading(disableBtn, defaultText);
```

Comment:
`setLoading` accepts 3 arguements. Please, pass the 3rd one in `finally` otherwise when I delete a card the confirmation button text becomes `Save` instead of `Yes`


```js
      popupInstance.setLoading(disableBtn, defaultText, defaultText);

```